StardewArchipelago/Locations/CodeInjections/Modded/SVE/BearShopStockModifier.cs

```diff
@@ -81,18 +81,22 @@ private StardewObject BerryIfChefsanityIsOn()
 
         private double BearDiscount()
         {
-            var hasKnowledge = _archipelago.HasReceivedItem("Bear Knowledge");
-            var knowledgeBuff = hasKnowledge ? 0.2f : 0f;
+            var hasKnowledge = _archipelago.HasReceivedItem("Bear's Knowledge");
+            var knowledgeBuff = hasKnowledge ? KNOWLEDGE_DISCOUNT : 0f;
             var applesHearts = 0;
             if (Game1.player.friendshipData.ContainsKey("Apples"))
             {
                 applesHearts = Game1.player.friendshipData["Apples"].Points / 250; // Get discount from being friends with Apples
             }
-            return INITIAL_DISCOUNT - (knowledgeBuff + applesHearts * APPLES_DISCOUNT);
+            return 1 - INITIAL_DISCOUNT - (knowledgeBuff + applesHearts * APPLES_DISCOUNT);

```

Comment:
i'm not sure I understand the math here.

`1 - INITIAL_DISCOUNT` -> This is the price after discount, right? But the method is called BearDiscount, not bearPrice
`[previous section] - (knowledgeBuff + applesHearts * APPLES_DISCOUNT)` -> This appears to **reduce** the discount, based on other discount sources. So the end result is a higher final value, the more discounts you have. This suggests that the methood **is** indeed a discount, but then what was up with the other discount?

Basically, you are discounting a discount, and I'm not sure exactly what is the intent here. I also notice that in many places where discount-type values are used, they are named "discount", and then used like "finalPrice = price * discount" which again suggests the reverse of a discount.

I would suggest going over all usages of price modifications and trying to normalize the names so that they say exactly what they do and overall makes this whole thing much easier to understand and parse

Reply:
So this is like, a definitional thing?  IE, instead of calling the method a discount, it should be a sale?  Something like that?